contribs/gmf/src/authentication/service.js

```diff
@@ -0,0 +1,242 @@
+goog.provide('gmf.authentication.service');
+
+goog.require('ngeo.CustomEvent');
+goog.require('gmf');
+goog.require('ol.events.EventTarget');
+
+// todo change test
+
+
+/**
+ * @typedef {{
+ *     functionalities: (gmfx.AuthenticationFunctionalities|undefined),
+ *     is_password_changed: (boolean|undefined),
+ *     role_id: (number|undefined),
+ *     role_name: (string|undefined),
+ *     username: (string|undefined)
+ * }}
+ */
+gmf.authentication.AuthenticationLoginResponse;
+
+
+/**
+ * @typedef {{
+ *     success: boolean
+ * }}
+ */
+gmf.authentication.AuthenticationDefaultResponse;
+
+
+/**
+ * @enum {string}
+ */
+gmf.authentication.AuthenticationRouteSuffix = {
+  CHANGE_PASSWORD: 'loginchange',
+  IS_LOGGED_IN: 'loginuser',
+  LOGIN: 'login',
+  LOGOUT: 'logout',
+  RESET_PASSWORD: 'loginresetpassword'
+};
+
+// todo?
+gmf.module.value('gmfUser', {
```

Comment:
Looks not suitable.
@gberaudo We move that in another file ?

Reply:
I do not understand why `gmfUser` is a standalone value instead of being a property of the authentication service.
To keep things simple I would let the value in this file (but register it into the authentication module).

Alternatives are:
- move it to its dedicated file, with its own Angular module and depend on it everywhere.
- make it a property of the authentication service.